src/Controllers/controller.py

```diff
@@ -20,6 +20,8 @@ def __init__(self):
         self._diagram = UML_Diagram()
         self._states = UML_States(self._diagram)
         self._should_quit = False
+        if isinstance(self._controller, CLI_Controller):
```

Comment:
instead of adding this here, add it in the pick controller method. Create the CLI controller with the copy of the diagram (that will be included in the __init__ method of the CLI_Controller) if you can't manage to make the Diagram singleton (which is the correct choice).

Reply:
diagram etc. is not defined during controller selection. Leaving it here for now.

Reply:
the diagram is created during UML Controller init. You can pass it out of that constructor into CLI_Controller's constructor by passing it into the pick controller method and moving the controller selection below the diagram creation.

Reply:
(the reason to change this is to get rid of the set diagram method, which is still a massive design mistake). It doesn't need to happen, but it would be good to.